src/Pim/Bundle/CatalogBundle/ElasticSearch/Filter/AbstractFilter.php

```diff
@@ -0,0 +1,129 @@
+use Pim\Component\Catalog\Validator\AttributeValidatorHelper;
+
+abstract class AbstractFilter implements FilterInterface
+{
+    /** @var SearchQueryBuilder */
+    protected $searchQueryBuilder = null;
+
+    /** @var array */
+    protected $supportedOperators = [];
+
+    /** @var AttributeValidatorHelper */
+    protected $attrValidatorHelper;
+
+    /** @var string[] */
+    protected $supportedAttributeTypes;
+
+    /** @var array */
+    protected $supportedFields;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsOperator($operator)
+    {
+        return in_array($operator, $this->supportedOperators);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOperators()
+    {
+        return $this->supportedOperators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAttributeTypes()
+    {
+        return $this->supportedAttributeTypes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsAttribute(AttributeInterface $attribute)
+    {
+        return in_array($attribute->getAttributeType(), $this->supportedAttributeTypes);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsField($field)
+    {
+        return in_array($field, $this->supportedFields);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFields()
+    {
+        return $this->supportedFields;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setQueryBuilder($searchQueryBuilder)
+    {
+        if (!$searchQueryBuilder instanceof SearchQueryBuilder) {
+            throw new \InvalidArgumentException(
+                'Query builder should be an instance of "\Pim\Bundle\CatalogBundle\ElasticSearch\SearchQueryBuilder"'
+            );
+        }
+
+        $this->searchQueryBuilder = $searchQueryBuilder;
+    }
+
+    /**
+     * Check locale and scope are valid
+     *
+     * @param AttributeInterface $attribute
+     * @param string             $locale
+     * @param string             $scope
+     *
+     * @throws InvalidPropertyException
+     */
+    protected function checkLocaleAndScope(AttributeInterface $attribute, $locale, $scope)
+    {
+        try {
+            $this->attrValidatorHelper->validateLocale($attribute, $locale);
+            $this->attrValidatorHelper->validateScope($attribute, $scope);
+        } catch (\LogicException $e) {
+            throw InvalidPropertyException::expectedFromPreviousException(
+                $attribute->getCode(),
+                static::class,
+                $e
+            );
+        }
+    }
+
+    /**
+     * Calculates the ES path to a product value indexed in ES.
+     *
+     * @param AttributeInterface $attribute
+     * @param string             $locale
+     * @param string             $scope
+     *
+     * @return string
+     */
+    protected function getAttributePath(AttributeInterface $attribute, $locale, $scope)
```

Comment:
OK for the moment, we'll rework this part later :)
we have to think about the way we will index products